tests/contrib/hooks/test_redis_hook.py

```diff
@@ -41,17 +41,33 @@ def test_get_conn_password_stays_none(self):
 
     def test_real_ping(self):
         hook = RedisHook(redis_conn_id='redis_default')
-        redis = hook.get_conn()
 
-        self.assertTrue(redis.ping(), 'Connection to Redis with PING works.')
+        self.assertTrue(hook.ping_redis(), 'Connection to Redis with PING works.')
 
     def test_real_get_and_set(self):
         hook = RedisHook(redis_conn_id='redis_default')
-        redis = hook.get_conn()
 
-        self.assertTrue(redis.set('test_key', 'test_value'), 'Connection to Redis with SET works.')
-        self.assertEqual(redis.get('test_key'), b'test_value', 'Connection to Redis with GET works.')
-        self.assertEqual(redis.delete('test_key'), 1, 'Connection to Redis with DELETE works.')
+        self.assertTrue(hook.set_key_value('test_key', 'test_value'), 'Connection to Redis with SET works.')
+        self.assertEqual(hook.get_key('test_key'), b'test_value', 'Connection to Redis with GET works.')
+        self.assertEqual(hook.delete_key('test_key'), 1, 'Connection to Redis with DELETE works.')
+
+    def test_exists(self):
+        hook = RedisHook(redis_conn_id='redis_default')
+
+        k, v = 'test_key_exists_true', 'test_val'
+
+        hook.set_key_value(k, v)
+
+        assert hook.check_if_key_exists(k)
+
+        hook.delete_key(k)
```

Comment:
I feel that this sort of test is closer to testing redis itself (though it does follow the pattern of the current behaviour)

We should probably go through all of the tests and work what we are over-testing/testing another project instead of our own and replace it with mocks.

Reply:
ok, mocked in hook tests for starters